### ISSUE TITLE
markDisabled does not trigger invalid state

### ISSUE THREAD

**Issue by rafaelucena** [ng-bootstrap/ng-bootstrap#3182]
### Bug description:
Using markDisabled, we set the date on the calendar to not be clickable, but it does not prevent the user to type the date by hand.
But, when using minDate or maxDate, everything works as expected.

The state returned when setting a date out of the range is invalid, as using, for example, day 31 of February.

### Link to minimally-working StackBlitz that reproduces the issue:
https://stackblitz.com/edit/angular-4mylnm

### Versions of Angular, ng-bootstrap and Bootstrap:
Angular: 7

ng-bootstrap: 4.1.2

**Comment by peterblazejewicz** [comment]
Only the *isValid* date and within min/max range conditions are tested. The `markDisabled` is for navigation purposes only (as per docs).
See this PR for upcoming changes to plug your logic from `markDisabled` into validation:
https://github.com/ng-bootstrap/ng-bootstrap/pull/3126

**Comment by rafaelucena** [comment]
Well, even if for navigation only... there isn't much use in disabling one date to have the user type it anyway.

Thank you for the link though. :)

**Comment by rafaelucena** [comment]
@benouat, I would like to ask for a revision of this situation.

Can this be considered a bug?

**Comment by benouat** [comment]
@rafaelucena I agree that something is not right here. I am still not convinced that we should flag it as bug...

A bug is a faulty behaviour of an existing feature. Here the feature to properly disabled some dates (independently of the current displayed month, and not as only visually, for current month, which `markDisabled` is doing) is not there at all.

Let me discuss it internally before moving on.

**Comment by rafaelucena** [comment]
@benouat, I understand.

Though I would like you to consider this also from the perspective of the usage of it.
Considering that the date might be disabled for whatever reason, it shouldn't make sense that setting that date by hand would be considered valid.

That would then force the developer to create a second validation internally to check if the date is disabled.

But, still, please let me know how the discussion went. Thank you :)

**Comment by danielczestki** [comment]
@rafaelucena Mayby You could fix it by your own in just fore easy steps:
1. fork
2. commit
3. push
4. merge request

;)

**Comment by danielczestki** [comment]
> The `markDisabled` is for navigation purposes only (as per docs).
> #3126

Then this should not be called markDisabled. Intuition tells you that this method do something different

**Comment by maxokorokov** [comment]
Also see potentially related #3126, #3215, #1325
